--- arnold/src/unix/main.c.orig	Sun Jun 15 20:11:53 2003
+++ arnold/src/unix/main.c	Sun Jun 15 20:12:24 2003
@@ -27,7 +27,7 @@
 #ifdef HAVE_SDL
 #include <SDL.h>
 #endif
-
+#define DATADIR "/usr/share/arnold"
 #include "display.h"
 #include <string.h>
 #include <unistd.h>
@@ -61,9 +61,8 @@
 
 	{
 		char LocalDirectory[1024];
-		char ProgramDirectory[1024]="";
+		char ProgramDirectory[1024]=DATADIR;
 		
-		getcwd(ProgramDirectory, 1024);
 
 sprintf(LocalDirectory,"%s/roms/amsdose/",ProgramDirectory);
 	
@@ -89,6 +88,7 @@
 	SetDirectoryForLocation(EMULATOR_ROM_KCCOMPACT_DIR,
 LocalDirectory);	
 
+	getcwd(ProgramDirectory,1024);
 
 	GenericInterface_Initialise();
 
